### PR TITLE
Added better validation for entry in option schema

### PR DESCRIPTION
**Pull request by halfzebra**
**What kind of change does this PR introduce?**
bugfix

**Did you add tests for your changes?**
Added one test

**If relevant, link to documentation update:**
N/A

**Summary**
#3534 The PR prevents users from having empty entries for bundles in the configuration file.

**Does this PR introduce a breaking change?**
No

**Other information**

### CODE REVIEW

**Comment by SpaceK33z** [comment]
It seems that the CI fails because of this change.

**Comment by halfzebra** [comment]
@SpaceK33z I'm working on the fix!

**Comment by halfzebra** [comment]
@SpaceK33z I have managed to fix it and add one more test to check the new validation error message.

File: test/Validation.test.js [hunk 1]
```diff
@@ -146,17 +166,20 @@ describe("Validation", function() {
...
-			} catch(e) {
-				if(!(e instanceof WebpackOptionsValidationError))
-					throw e;
-				e.message.should.startWith("Invalid configuration object.");
-				e.message.split("\n").slice(1).should.be.eql(testCase.message);
-				return;
-			}
-			throw new Error("Validation didn't fail");
-		})
+		if(testCase.name === "empty entry bundle array") {
```
Comment (sokra) [inline test/Validation.test.js]: You can run single tests with `mocha -g "empty entry bundle array"`

**Comment by sokra** [comment]
Looks good if you remove the test typo.

File: test/Validation.test.js (hunk 1 above)
Comment (halfzebra) [inline test/Validation.test.js]: @sokra my bad! Thanks for the suggestion, I have removed the leftovers from testing.

Comment (TheLarkInn) [inline test/Validation.test.js]: No worries I'll merge once CI clears

**Comment by TheLarkInn** [comment]
Congrats on merging your first webpack PR. Thank you for the contribution. 🎉 🍾

**Comment by halfzebra** [comment]
@TheLarkInn Thanks for the help! 😃

**Comment by awinograd** [comment]
This ended up being a breaking change for me. Using a custom dev server & middleware to start up the dev server without building any bundles and then build them as they get requested. 

Totally understand the desire for strict validation (which has definitely saved me several times w/webpack) but maybe worth adding a release note?

**Pull request merged**